reference-apps/aaos-unbundled/mediacompose/src/main/java/com/android/designcompose/reference/mediacompose/MediaAdapter.kt

```diff
@@ -326,20 +329,31 @@ class MediaAdapter(
             return
         }
         when (newBrowsingState.mConnectionStatus) {
-            MediaBrowserConnector.ConnectionStatus.CONNECTING -> {}
+            MediaBrowserConnector.ConnectionStatus.CONNECTING -> {
+                canSearch.value = false
+            }
             MediaBrowserConnector.ConnectionStatus.CONNECTED -> {
                 Log.i(
                     TAG,
                     "Media Browse State CONNECTED ${newBrowsingState.mMediaSource.getDisplayName(context)}"
                 )
                 val browser = newBrowsingState.mBrowser
                 canSearch.value = MediaBrowserViewModelImpl.getSupportsSearch(browser)
+
+                val rootId: String? = browser?.root
+                mediaRepo.getMediaChildren(rootId, Bundle()).observeForever { items ->
+                    browseStack.resetTo(
+                        if (items?.data?.isEmpty() == true) null else items?.data?.get(0)
+                    )
+                }
             }
             MediaBrowserConnector.ConnectionStatus.DISCONNECTING,
             MediaBrowserConnector.ConnectionStatus.REJECTED,
             MediaBrowserConnector.ConnectionStatus.SUSPENDED -> {
                 browseStack.clear()
+                canSearch.value = false
             }
+            MediaBrowserConnector.ConnectionStatus.NONEXISTENT -> {}
```

Comment:
Should it be grouped with DISCONNECTING, REJECTED and SUSPENDED? I'm just not sure what NONEXISTENT means.

Reply:
MediaBrowserConnector has a connectTo(MediaSource) function that it will either enter CONNECTING state or NONEXISTENT state(means the MediaSource doesn't exist). I think in this case it should not do anything.